LibThreatClassic2.lua

```diff
@@ -1175,25 +1457,94 @@ do
 			self:SendCommRaw(self:GroupDistribution(), "THREAT_UPDATE", petMsg)
 		end
 		lastPublishTime = GetTime()
+
+		self:PublishMeleeRange(force)
+	end
+end
+
+-- #NODOC
+do
+	local t = {}
+
+	local function getMeleeRangeString(force)
+		local data = meleeRange[playerGUID]
+
+		if not data then
+			return nil, false
+		end
+
+		local nl = 0
+		local changed = false
+
+		for k, v in pairs(data) do
+			if lastPublishedMeleeRange[k] ~= v or force then
+				nl = nl + 1
+
+				if v then
+					t[nl] = format("%s=%d,", k, v and 1 or 0)
+				else
+					t[nl] = format("%s=,", k)
+				end
+
+				lastPublishedMeleeRange[k] = v
+				changed = true
+			end
+		end
+
+		for k, v in pairs(lastPublishedMeleeRange) do
+			if data[k] == nil then
+				nl = nl + 1
+				t[nl] = format("%s=,", k)
+				lastPublishedMeleeRange[k] = nil
+				changed = true
+			end
+		end
+
+		if changed then
+			return tconcat(t, "", 1, nl), true
+		else
+			return nil, false
+		end
+	end
+
+	-- #NODOC
+	function ThreatLib:PublishMeleeRange(force)
+		if (not inParty and not inRaid) or self.dontPublishMeleeRange then return end
+
+		local msg = getMeleeRangeString(force)
+
+		if msg then
+			self:SendCommRaw(self:GroupDistribution(), "MELEE_RANGE_UPDATE", msg)
+		end
+
+		lastPublishMeleeRangeTime = GetTime()
 	end
 end
 
 -- #NODOC
 function ThreatLib:GetPublishInterval()
-	-- Scale publish interval from 1.0 to 1.5 based on party size, half that for tanks
-	-- We'll be at 1.5 sec for 0-5 party size, scale from 1.5 to 2.5 for 6-25 players, and stay at 2.5 for > 20 players
-	-- This means that we'll transmit less data in a raid
-	local playerClass = playerClass or select(2, UnitClass("player"))
-	local partyNum = max(0, (self.currentPartySize or 0) - 5)
-	local interval = min(1.5, 1 + (1 * (partyNum / 20)))
+	local role = self:GetRole(playerGUID)
+	local interval = 1.5
 
-	-- Make tanks update more often
-	if playerClass == "WARRIOR" or playerClass == "DRUID" or playerClass == "PALADIN" then
-		interval = interval * 0.5
+	if role == "TANK" then
+		interval = 0.75
+	elseif role == "HEALER" then
+		interval = 3.0
 	end
+
 	return interval
 end
 
+-- #NODOC
+function ThreatLib:GetPublishMeleeRangeInterval()
+	return 0.5
```

Comment:
Also this value should be much higher. At 800 CPS throttle and 39 characters per target (36 targetguid, 1 true/false, = and ,) plus the playerguid and message name, moving in and out of Garr adds range will instantly trigger throttle.

This should trigger every 5 seconds at most. Especially because it triggers instant on the first change.